sdk/python/lib/pulumi/errors.py

```diff
@@ -12,9 +12,40 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from typing import List, Optional, TypedDict
+
 
 class RunError(Exception):
     """
     Can be used for terminating a program abruptly, but resulting in a clean exit rather than the usual
     verbose unhandled error logic which emits the source program text and complete stack trace.
     """
+
+
+class InputPropertyError(Exception):
+    def __init__(self, message: str, property_path: str, reason: str):
+        """
+        Can be used to indicate that the client has made a request with a bad input property.
+        """
+        self.message = message
+        self.property_path = property_path
+        self.reason = reason
+
+
+class PropertyError(TypedDict):
```

Comment:
Should we use a suffix other than `Error` for this? Right now it looks like it's another exception error that can be thrown (based on the name). Perhaps `InputPropertyErrorDetails` to match Node.js?

Reply:
Good idea, done :heavy_check_mark: